data_management/opensearch/index_file_content_and_metadata_in_opensearch.py

```diff
@@ -0,0 +1,131 @@
+    file_data: Dict[str, Any], file_stream: bytes
+) -> Dict[str, Any]:
+    file_type = file_data["file_name"].split(".")[-1].lower()
+    new_file_data = file_data
+    if file_type in ["txt", "docx", "pdf"]:
+        new_file_data["content"] = extract_text(file_stream)
+    return new_file_data
+
+
+def _fetch_file_data(
+    file_id: str, database_url: str
+) -> Dict[str, Union[str, Dict[str, str]]]:
+    engine = create_engine(database_url)
+    Base = declarative_base()
+    Base.metadata.reflect(bind=engine)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    query = """
+    SELECT
+        f."FileId" AS file_id,
+        f."FileName" AS file_name,
+        f."FileReference" AS file_reference,
+        f."FilePath" AS file_path,
+        f."CiteableReference" AS citeable_reference,
+        s."SeriesId" AS series_id,
+        s."Name" AS series_name,
+        b."Name" AS transferring_body,
+        b."BodyId" AS transferring_body_id,
+        b."Description" AS transferring_body_description,
+        c."ConsignmentReference" AS consignment_reference,
+        fm."PropertyName",
+        fm."Value"
+    FROM
+        "File" f
+    JOIN
+        "Consignment" c ON f."ConsignmentId" = c."ConsignmentId"
+    JOIN
+        "Series" s ON c."SeriesId" = s."SeriesId"
+    JOIN
+        "Body" b ON s."BodyId" = b."BodyId"
+    LEFT JOIN
+        "FileMetadata" fm ON f."FileId" = fm."FileId"
+    WHERE
+        f."FileId" = :file_id;
+    """
+    result = session.execute(text(query), {"file_id": file_id}).fetchall()
+
+    session.close()
+
+    file_data = {
+        "file_id": result[0].file_id,
+        "file_name": result[0].file_name,
+        "file_reference": result[0].file_reference,
+        "file_path": result[0].file_path,
+        "citeable_reference": result[0].citeable_reference,
+        "series_id": result[0].series_id,
+        "series_name": result[0].series_name,
+        "transferring_body": result[0].transferring_body,
+        "transferring_body_id": result[0].transferring_body_id,
+        "transferring_body_description": result[
+            0
+        ].transferring_body_description,
+        "consignment_reference": result[0].consignment_reference,
+        "metadata": {
+            row.PropertyName: row.Value
+            for row in result
+            if row.PropertyName is not None
+        },
+    }
+
+    return file_data
+
+
+def extract_text(file_stream: bytes) -> str:
+    with tempfile.NamedTemporaryFile(delete=True) as temp:
+        temp.write(file_stream)
+        temp.flush()
+        context = textract.process(temp.name)
+    return context.decode("utf-8")
+
+
+def _index_in_opensearch(
+    file_id: str,
+    document: Dict[str, Any],
+    open_search_host_url: str,
+    open_search_http_auth: AWS4Auth,
+) -> None:
+    open_search = OpenSearch(
+        open_search_host_url,
+        http_auth=open_search_http_auth,
+        use_ssl=True,
+        verify_certs=False,
```

Comment:
ideally we would verify certs. Might make a followup ticket to deal with this, as we also need to handle for the webapp when we search.

Reply:
just switched to True now.